src/Spice86.Core/Emulator/OperatingSystem/DosPathResolver.cs

```diff
@@ -0,0 +1,236 @@
+namespace Spice86.Core.Emulator.OperatingSystem;
+
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+using Spice86.Core.Emulator.OperatingSystem.Structures;
+using Spice86.Shared.Emulator.Errors;
+using Spice86.Shared.Interfaces;
+using Spice86.Shared.Utils;
+
+/// <inheritdoc />
+public class DosPathResolver : IDosPathResolver {
+
+    /// <inheritdoc />
+    public IDictionary<char, MountedFolder> DriveMap { get; private set; } = new Dictionary<char, MountedFolder>();
+
+    /// <inheritdoc />
+    public char CurrentDrive { get; private set; }
+
+    /// <inheritdoc />
+    public string CurrentHostDirectory {
+        get => DriveMap[CurrentDrive].CurrentFolder;
+        private set => DriveMap[CurrentDrive].CurrentFolder = value;
+    }
+
+    private readonly ILoggerService _loggerService;
+
+    /// <summary>
+    /// Initializes a new instance.
+    /// </summary>
+    /// <param name="loggerService">The logger service implementation.</param>
+    public DosPathResolver(ILoggerService loggerService) => _loggerService = loggerService;
+
+    /// <inheritdoc />
+    public string GetHostRelativePathToCurrentDirectory(string path) => Path.GetRelativePath(CurrentHostDirectory, path);
+
+    /// <inheritdoc/>
+    public DosFileOperationResult SetCurrentDir(string newCurrentDir) {
+        if (IsPathRooted(newCurrentDir)) {
+            string? newCurrentDirectory = ToHostCaseSensitiveFullName(newCurrentDir, false);
+            if (!string.IsNullOrWhiteSpace(newCurrentDirectory)) {
+                CurrentHostDirectory = newCurrentDirectory;
+                return DosFileOperationResult.NoValue();
+            }
+        }
+
+        if (newCurrentDir == ".") {
+            return DosFileOperationResult.NoValue();
+        }
+
+        if (newCurrentDir == "..") {
+            CurrentHostDirectory = GetFullNameForParentDirectory(CurrentHostDirectory) ?? CurrentHostDirectory;
+            return DosFileOperationResult.NoValue();
+        }
+
+        while (newCurrentDir.StartsWith("..\\")) {
+            newCurrentDir = newCurrentDir[3..];
+            CurrentHostDirectory = GetFullNameForParentDirectory(CurrentHostDirectory) ?? CurrentHostDirectory;
```

Comment:
GetFullNameForParentDirectory never returns null so the ?? does nothing

Reply:
Fixed.